### ISSUE TITLE
[BUG] Tool Agent doesnt have usedTools property in the response.

### ISSUE THREAD

**Issue by emrahtoy** [FlowiseAI/Flowise#2575]
**Describe the bug**
OpenAI Tool Agent is depreciating and there is a new Tool Agent in exchange.

OpenAI Tool Agent has usedTools property in the response but Tool Agent does not have it.

**To Reproduce**

1. Change OpenAI Tool Agent with Tool Agent component
2. Run the flow and await for the rest API response.
3. Check the result
![image](https://github.com/FlowiseAI/Flowise/assets/4387637/0e3e4284-1184-4e56-beb5-ae7417011372)
4. There is no usedTools property.

**Expected behavior**
Tool Agent also must have usedTools property in its response.

**Flow**
![image](https://github.com/FlowiseAI/Flowise/assets/4387637/adac1eee-c8b1-4bb3-806b-654bfd0fbbc8)

**Comment by HenryHengZJ** [comment]
Pretty sure it does:

![image](https://github.com/FlowiseAI/Flowise/assets/26460777/e35a48f9-c10a-4105-837a-894a74fb1542)
![image](https://github.com/FlowiseAI/Flowise/assets/26460777/af5cd5bd-4110-4f2c-8498-42ddfdc34eb3)

**Comment by emrahtoy** [comment]
Thank you for your quick response @HenryHengZJ i will try another function. It seems like there is a chance that my function may have never been called.

**Comment by emrahtoy** [comment]
I have been using tool calls to get the proper JSON format in the response without returning anything from the function. In the past, this was known as a tricky solution to get formatted JSON in the stream. Right now it is not working as I have to return a result from the function I called and only then it is working as intended otherwise it just ignores the function call.
I should have seen this difference before, I thank you for your precious help @HenryHengZJ 

Closing as it is clear to me now.
